provider/import.go

```diff
@@ -149,10 +153,15 @@ func Import(ctx context.Context, p Provider, hcl, tfstate writer.Writer, f *filt
 	}
 
 	if hcl != nil {
+		if _, ok := hcl.(*h.Writer); ok {
+			err := util.Interpolate(hcl.(*h.Writer), interpolation)
+			if err != nil {
+				return errors.Wrapf(err, "error while Interpolate Config")
+			}
+		}
```

Comment:
When I was reading the `util.Interpolate` I already thought that it should go to the `hcl` pkg, and now even more so the implementation is the one that tries to convert to the specific type even with that I don't like that as the Interface was ment to avoid knowing the implementation behind it, and this brakes it.

Ways to solve this:
* Have the `util` be the one that converts so it expects a `write.Writer` not an `hcl.Writer` (still not good IMO)
* Move it to `hcl` pkg
* Add it to the `write.Wiriter` interface
* Make the `Sync()` expect the `map[string]string`.

The last 2 are the ones I prefer as they delay the implementation to the actual one (State will do nothing as no Interpolation is needed and HCL will do something).